ecl/hql/hqlgram.y

```diff
@@ -7308,6 +7308,7 @@ abstractModule
                         }
     | IF '(' booleanExpr ',' abstractModule ',' abstractModule ')'
                         {
+                            parser->checkConstant($3, true);
```

Comment:
shouldn't this be callAllowed=> false? i.e shouldn't it be checkConstant($3, false)

Reply:
No, calls are allowed, as long as they get constant folded later.  The default is to pass true. It looks like false is only ever passed when it is a parameter to an attribute on a field (e.g., maxcount) - probably because that causes problems with --fastsyntax